lib/tapioca/runtime/reflection.rb

```diff
@@ -255,6 +255,21 @@ def child_module_for_parent_with_name(parent, name)
       def method_defined_by_forwardable_module?(method)
         method.source_location&.first == Object.const_source_location(:Forwardable)&.first
       end
+
+      sig { params(name: String).returns(T::Boolean) }
+      def has_aliased_namespace?(name)
+        name_parts = name.split("::")
+        name_parts.pop # drop the constant name, leaving just the namespace
+
+        name_parts.each_with_object([]) do |name_part, namespaces|
+          namespaces << "#{namespaces.last}::#{name_part}".delete_prefix("::")
+        end.any? do |namespace|
+          constant = constantize(namespace)
+          next unless Module === constant
+
+          name_of(constant) != namespace
```

Comment:
Should we extract this as a private `constant_name_differs_from_actual_namespace` method to facilitate reading? Maybe just adding a comment in the block would help.